nanoFramework.System.Net/NetworkHelper/NetworkHelper.cs

```diff
@@ -0,0 +1,300 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Diagnostics;
+using System.Net.NetworkInformation;
+using System.Threading;
+
+namespace nanoFramework.Networking
+{
+    /// <summary>
+    /// Network helper class providing helper methods to assist on connecting to a network.
+    /// </summary>
+    public static class NetworkHelper
+    {
+        private static ManualResetEvent _ipAddressAvailable;
+        private static ManualResetEvent _networkReady;
+
+        private static bool _requiresDateTime;
+        private static IPConfiguration _ipConfiguration;
+        private static NetworkHelperStatus _networkHelperStatus = NetworkHelperStatus.None;
+
+        // defaulting to Ethernet
+        internal static NetworkInterfaceType _workingNetworkInterface = NetworkInterfaceType.Ethernet;
+
+        /// <summary>
+        /// This flag will make sure there is only one and only call to any of the helper methods.
+        /// </summary>
+        internal static bool _helperInstanciated = false;
+
+        /// <summary>
+        /// Event signaling that networking it's ready.
+        /// </summary>
+        /// <remarks>
+        /// The conditions for this are setup in the call to <see cref="SetupNetworkHelper"/>. 
+        /// It will be a composition of network connected, IpAddress available and valid system <see cref="DateTime"/>.</remarks>
+        public static ManualResetEvent NetworkReady => _networkReady;
+
+        /// <summary>
+        /// Status of NetworkHelper.
+        /// </summary>
+        public static NetworkHelperStatus Status => _networkHelperStatus;
+
+        /// <summary>
+        /// This method will setup the network configurations so that the <see cref="NetworkReady"/> event will fire when the required conditions are met.
+        /// That will be the network connection to be up, having a valid IpAddress and optionally for a valid date and time to become available.
+        /// </summary>
+        /// <param name="requiresDateTime">Set to <see langword="true"/> if valid date and time are required.</param>
+        /// <exception cref="InvalidOperationException">If any of the <see cref="NetworkHelper"/> methods is called more than once.</exception>
+        /// <exception cref="NotSupportedException">There is no network interface configured. Open the 'Edit Network Configuration' in Device Explorer and configure one.</exception>
+        public static void SetupNetworkHelper(bool requiresDateTime = false)
```

Comment:
Same question here, why this one while you have the constructor with the cancellation token?

Reply:
Same answer 😅